tests/test_recommendation.py

```diff
@@ -1,14 +1,107 @@
 # -*- coding: utf-8 -*-
 # vim:ts=4:sw=4:expandtab
 
-from muacrypt.recommendation import Recommendation
+from muacrypt import mime
+
+
+def send_ac_mail(sender, recipient, Date=None):
+    mail = mime.gen_mail_msg(
+        From=sender.addr, To=[recipient.addr],
+        Autocrypt=sender.make_ac_header(recipient.addr),
+        Date=Date,
+    )
+    recipient.process_incoming(mail)
+
+def send_enc_ac_mail(sender, recipient):
+    msg = mime.gen_mail_msg(
+        From=sender.addr, To=[recipient.addr],
+        Autocrypt=sender.make_ac_header(recipient.addr),
+    )
+    r = sender.encrypt_mime(msg, [recipient.addr])
+    recipient.process_incoming(r.enc_msg)
+    r = recipient.decrypt_mime(r.enc_msg)
+    return r.dec_msg
+
+def send_no_ac_mail(sender, recipient):
+    mail = mime.gen_mail_msg(
+        From=sender.addr, To=[recipient.addr],
+    )
+    recipient.process_incoming(mail)
+
+
+def get_recommendation(composer, peer, reply_to_enc=False):
+    return composer.get_recommendation([peer.addr],
+            reply_to_enc=reply_to_enc)
 
 
 class TestRecommendation:
 
-    def test_empty_peer_state(self, account_maker):
-        sender, recipient = account_maker(), account_maker()
-        peerstate = sender.get_peerstate(recipient.addr)
-        rec = Recommendation({recipient.addr: peerstate})
-        assert rec.target_keys()[recipient.addr] is None
+    def test_disable_on_initial_mail(self, account_maker):
+        composer, peer = account_maker(), account_maker()
+        rec = get_recommendation(composer, peer)
+        assert rec.target_keys()[peer.addr] is None
         assert rec.ui_recommendation() == 'disable'
+
+    def test_available_after_receiving_ac_mail(self, account_maker):
+        composer, peer = account_maker(), account_maker()
+        send_ac_mail(peer, composer)
+        rec = get_recommendation(composer, peer)
+        assert rec.target_keys()[peer.addr]
+        assert rec.ui_recommendation() == 'available'
+
+    def test_disable_after_receiving_no_ac_mail(self, account_maker):
+        composer, peer = account_maker(), account_maker()
+        send_no_ac_mail(peer, composer)
+        rec = get_recommendation(composer, peer)
+        assert rec.target_keys()[peer.addr] is None
+        assert rec.ui_recommendation() == 'disable'
+
+    def test_available_long_after_receiving_ac_mail(self, account_maker):
+        long_ago = 'Sun, 15 Jan 2017 15:00:00 -0000'
+        composer, peer = account_maker(), account_maker()
+        send_ac_mail(peer, composer, Date=long_ago)
+        rec = get_recommendation(composer, peer)
+        assert rec.target_keys()[peer.addr]
```

Comment:
this and the other "key" asserts could check that the target key is ``== peer.public_keyhandle``.  Maybe it should be ``rec.target_keyhandles`` maybe?